src/components/tiles/text/text-toolbar.tsx

```diff
@@ -81,85 +122,66 @@ export const TextToolbarComponent: React.FC<IProps> = (props: IProps) => {
         break;
       case "italic":
-        editor.command("toggleMark", EFormat.italic);
+        toggleMark(editor, EFormat.italic);
         break;
       case "underline":
-        editor.command("toggleMark", EFormat.underlined);
+        toggleMark(editor, EFormat.underlined);
         break;
       case "subscript":
-        handleToggleSuperSubscript(EFormat.subscript, editor);
+        toggleSuperSubscript(editor, EFormat.subscript);
         break;
       case "superscript":
-        handleToggleSuperSubscript(EFormat.superscript, editor);
+        toggleSuperSubscript(editor, EFormat.superscript);
         break;
       case "list-ol":
-        editor.command("toggleBlock", EFormat.numberedList);
+        toggleBlock(editor, EFormat.numberedList);
         break;
       case "list-ul":
-        editor.command("toggleBlock", EFormat.bulletedList);
+        toggleBlock(editor, EFormat.bulletedList);
         break;
       case "undo":
         editor.undo();
         break;
       default: {
         const toolInfo = getTextPluginInfo(buttonIconName);
-
         // Handle Text Plugins
-        if (!toolInfo?.command) {
-          console.warn("Can't find text plugin command for", buttonIconName);
+        if (!toolInfo || !pluginModalHandlers[toolInfo.iconName]) {
+          console.warn("Can't find text plugin handler for", buttonIconName);
           break;
         }
-        // Send the dialogController to all plugins
-        //
-        // TODO: I think this should be an object: `{dialogController}`
-        // instead of a raw param. This way we can add more props to it
-        // without changing the method signature and worrying about argument
-        // order. The reason is that I hope we can provide additional
-        // controllers or services that plugins can use. This change should be
-        // made in slate-editor too for consistency.
-        editor.command(toolInfo?.command, dialogController);
+        pluginModalHandlers[toolInfo.iconName]();
       }
     }
   };
 
-  // listen for configuration requests from plugins
-  useEffect(() => {
-    const emitter: EventEmitter | undefined = editor?.query("emitter");
-    const handler = (event: string, ...args: any) => {
-      editor?.command(event, dialogController, ...args);
-    };
-    emitter?.on("toolbarDialog", handler);
-    return () => {
-      emitter?.off("toolbarDialog", handler);
-    };
-  }, [editor, dialogController]);
-
   return documentContent
     ? ReactDOM.createPortal(
         <div className={`text-toolbar ${enabled && toolbarLocation ? "enabled" : "disabled"}`}
               style={toolbarLocation} onMouseDown={handleMouseDown}>
           {toolbarButtons.map(button => {
-            const { iconName, Icon, toolTip } = button;
+            const { iconName, Icon, toolTip, buttonEnabled } = button;
+            let bEnabled = enabled;
+            if (buttonEnabled) {
+              bEnabled = buttonEnabled(selectedVariable);
+            }
```

Comment:
This is another bit that needs to be extracted into the plugin so there isn't a direct dependency on the variables stuff.